Makefile

```diff
@@ -1,37 +1,61 @@
-.ONESHELL:
 .PHONY:
 
 all: install
 
 ##### Variables ######
 COLOR := "\e[1;36m%s\e[0m\n"
 
+# Disable cgo by default.
+CGO_ENABLED ?= 0
+
 TEMPORAL_ROOT := temporal
 TCTL_ROOT := tctl
+CLI_ROOT := cli
+DOCKERIZE_ROOT := dockerize
 IMAGE_TAG ?= sha-$(shell git rev-parse --short HEAD)
 TEMPORAL_SHA := $(shell sh -c 'git submodule status -- temporal | cut -c2-40')
 TCTL_SHA := $(shell sh -c "git submodule status -- tctl | cut -c2-40")
 
 DOCKER ?= docker buildx
 BAKE := IMAGE_TAG=$(IMAGE_TAG) TEMPORAL_SHA=$(TEMPORAL_SHA) TCTL_SHA=$(TCTL_SHA) $(DOCKER) bake
+NATIVE_ARCH := $(shell go env GOARCH)
 
 ##### Scripts ######
 install: install-submodules
+clean:
+	rm -rf ./build
 
 update: update-submodules
 
 install-submodules:
-	@printf $(COLOR) "Installing temporal and tctl submodules..."
-	git submodule update --init $(TEMPORAL_ROOT) $(TCTL_ROOT)
+	@printf $(COLOR) "Installing submodules..."
+	git submodule update --init
 
 update-submodules:
 	@printf $(COLOR) "Updatinging temporal and tctl submodules..."
 	git submodule update --force --remote $(TEMPORAL_ROOT) $(TCTL_ROOT)
 
 ##### Docker #####
 
-build:
-	$(BAKE)
+# If you're new to Make, this is a pattern rule: https://www.gnu.org/software/make/manual/html_node/Pattern-Rules.html#Pattern-Rules
+# $* expands to the stem that matches the %, so when the target is amd64-bins $* expands to amd64
+%-bins:
+	mkdir -p build/$*
+	cd $(DOCKERIZE_ROOT) && CGO_ENABLED=$(CGO_ENABLED) GOOS=linux GOARCH=$* go build -o dockerize .
```

Comment:
general concern: if we build outside of docker, we leak various details about the build environment into the binary, including the path that this repo is checked out to. is it possible to do all this but inside docker still?

Reply:
We could do this all within docker if we set up a tree-shaped pipeline with separate amd64 and aarch64 builders followed by a final step to combine the images but that's a larger change by far.

Ref: https://www.docker.com/blog/multi-arch-build-and-images-the-simple-way/

Reply:
I believe this is a, if not MUST, but very desirable option.

Reply:
For what reason? The configuration of our build environment (minus secrets) is entirely public

Reply:
We discussed this elsewhere: the big change is that when (or if) we need to build locally we're exposing details of a private machine.

It would be great to remove this problem in the future by using a proper arm64 runner, but I want to improve our build's reliability first so we don't need to build images manually in the first place